From 5e355e09ece32b10503352508745ca3ce5998796 Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:43:53 -0500
Subject: [wireless] cfg80211: don't WARN about two consecutive Country IE hint

Message-id: <1418208233-8520-1-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101608
O-Subject: [RHEL7.1 PATCH] cfg80211: don't WARN about two consecutive Country IE hint
Bugzilla: 1164282
RH-Acked-by: John Linville <linville@redhat.com>

BZ#1164282

Upstream status:
Patch was posted: http://marc.info/?l=linux-wireless&m=141750712606973&w=2

Description:
This can happen and there is no point in added more
detection code lower in the stack. Catching these in one
single point (cfg80211) is enough. Stop WARNING about this
case.

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Tested on wlan-r2s2.wlan.rhts.eng.bos.redhat.com where the problem
was reproducible.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/wireless/reg.c b/net/wireless/reg.c
index 14c86b4..9d164df 100644
--- a/net/wireless/reg.c
+++ b/net/wireless/reg.c
@@ -1799,11 +1799,8 @@ __reg_process_hint_country_ie(struct wiphy *wiphy,
    return REG_REQ_IGNORE;
   return REG_REQ_ALREADY_SET;
  }
- /*
-  * Two consecutive Country IE hints on the same wiphy.
-  * This should be picked up early by the driver/stack
-  */
- if (WARN_ON(regdom_changes(country_ie_request->alpha2)))
+
+ if (regdom_changes(country_ie_request->alpha2))
   return REG_REQ_OK;
  return REG_REQ_ALREADY_SET;
 }
-- 
1.7.1